src/mon/MDSMonitor.cc

```diff
@@ -988,12 +988,16 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
       ostringstream ess;
       mds_gid_t gid = gid_from_arg(fsmap, id, ess);
       if (gid == MDS_GID_NONE) {
-	// the mds doesn't exist, but no file systems are unhappy, so losing it
-	// can't have any effect.
-	continue;
+        invalid_ids.push_back(std::string(id));
+        continue;
       }
       stopping.insert(gid);
     }
+    if (!invalid_ids.empty()) {
+      r = -EINVAL;
+      ss << "invalid mds(s) specified: " + stringify(invalid_ids);
```

Comment:
instead of keeping the list and then serializing it here, I'd suggest that you simply copy the output from `ess` here 

```
      if (gid == MDS_GID_NONE) {
	// the mds doesn't exist, but no file systems are unhappy, so losing it
	// can't have any effect.
	ss << ess.str();
        rc = -ENOENT; // if we decide that it's OK to return an error, see my other comment
        continue;
      }
```